src/Spago/Env.hs

```diff
@@ -119,6 +126,7 @@ data InstallEnv = InstallEnv
   , envConfigPath :: !ConfigPath
   , envGlobalCache :: !GlobalCache
   , envPackageSet :: !PackageSet
+  , envPursCmd :: !PursCmd
```

Comment:
Adding the `purs` command to the `InstallEnv` is a bit smelly here (like, `install` should just "download packages", why do we need `purs` available for that?), so I've been wondering if there's any way to avoid it.
It looks like it's because `Spago.FetchPackage.fetchPackages` calls `PackageSet.checkPursIsUpToDate`. Maybe we should take away that check from there and just call it in all the places where `fetchPackages` is called?
I don't have a strong opinion on this though, maybe this is fine - I'm just voicing my reasoning to see if any of this makes sense to you

Reply:
Right, so the reason I originally moved `purs` into `InstallEnv` was a change that I made to `pursVersion`: I made it take in a reference to `purs` https://github.com/purescript/spago/pull/730/commits/7ff3fce7c10460ab229f73150780388693da731d#diff-f5cfb5054fe7614682b6a1be9572e2b1755df350ac60006146918c2df34afc41L131

I did this because I wasn't sure if using "purs" was always correct after seeing a few places reference "purs.cmd" for Windows. I've since reverted that change, it was working before so I shouldn't change it.

Since I've reverted that change, I can go through and restore the old behavior and then `purs` doesn't have to be stored in `InstallEnv`.

I didn't realize that the install continues if it can't verify the purs version against the package set - I thought it failed. Now that I know it just doesn't check it, I think you're right, it's a bit smelly to have `purs` be in `InstallEnv`. I'll try to untangle it